### PR TITLE
Index snapshots and refactoring [#175781663] [#175781589]

### PR DESCRIPTION
**Pull request by behrangi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -313,6 +313,16 @@
                 </plugin>
             </plugins>
         </pluginManagement>
+        <plugins>
```
Comment (awaisathar) [inline pom.xml:316]: Do we need this?

File: src/main/java/uk/ac/ebi/biostudies/api/util/SnapshotAwareDirectoryTaxonomyWriter.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class SnapshotAwareDirectoryTaxonomyWriter extends DirectoryTaxonomyWriter {
+
+    private SnapshotDeletionPolicy facetIndexSnapShot;
+
+    public SnapshotAwareDirectoryTaxonomyWriter(Directory facetDirectory, IndexWriterConfig.OpenMode openMode) throws IOException{
+            super(facetDirectory, openMode);
+    }
+
+    @Override
+    protected IndexWriterConfig	createIndexWriterConfig(IndexWriterConfig.OpenMode openMode){
```
Comment (awaisathar) [inline src/main/java/uk/ac/ebi/biostudies/api/util/SnapshotAwareDirectoryTaxonomyWriter.java:21]: Extra space between return type and method name. Let's start formatting the code to the default IntelliJ style

**Review by awaisathar — COMMENTED** [review]
LGTM. Please see the comments

File: pom.xml (hunk 1 above)
Comment (behrangi) [reply pom.xml:316]: It is required for java 11 features  but  if it creates problem I can use another instruction
FileSystemIndexTransferer -> line 52
        Path destination = Path.of(destDir);

File: src/main/java/uk/ac/ebi/biostudies/api/util/SnapshotAwareDirectoryTaxonomyWriter.java (hunk 2 above)
Comment (behrangi) [reply src/main/java/uk/ac/ebi/biostudies/api/util/SnapshotAwareDirectoryTaxonomyWriter.java:21]: This happened during merge ...

**Review by awaisathar — APPROVED** [review]

**Pull request merged**